h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -94,6 +101,74 @@ public void addScoringInfo(GLMParameters parms, int nclasses, long currTime, int
   
   public void setVcov(double[][] inv) {_output._vcov = inv;}
 
+  /***
+   * This method will calculate the variable inflation factor of each numerical predictor using the following
+   * procedure:
+   * 1. For each numerical predictor, choose it as the response variable and leave all the other predictors as the
+   * predictors;
+   * 2. Build a GLM model and obtained the R2 of the model;
+   * 3. the variable inflation factor for that predictor is calculated as 1.0/(1.0-R2).
+   * 
+   * All the variable inflation factors for all valid predictors are saved in a double array.  No variable inflation
+   * factors are generated for non-numerical predictors.
+   * 
+   */
+  public String[] buildVariableInflationFactors(Frame train, DataInfo dinfo) {
+    String[] predictorNames = extractPredictorNames(_parms, dinfo, _parms._fold_column);
+    // only calculate VIF for numerical columns
+    String[] vifPredictors = getVifPredictors(train, _parms, dinfo);
+    return buildVariableInflationFactors(_parms, vifPredictors, predictorNames);
+  }
+
+  static String[] getVifPredictors(Frame train, GLMParameters parms, DataInfo dinfo) {
+    String[] predictorNames = extractPredictorNames(parms, dinfo, parms._fold_column);
+    return Stream.of(predictorNames)
```

Comment:
predictorNames is already extracted in line 117, can you just pass it as an argument to getVifPredictors?

Reply:
there are 2 context in which the method is called - in one case we have it (line 117) in the init method validation we don't have it, I think we can keep it as-is - extracting the predictors is cheap (and quick)